src/api/SequenceNumber/useSequenceNumber.ts

```diff
@@ -0,0 +1,43 @@
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { subWebContext } from "src/api/SPWebContext";
+import { z } from "zod";
+import "@pnp/sp/items";
+
+const SequenceOffset = z.array(
+  z.object({
+    Id: z.number().int().positive(),
+    Title: z.number().int().positive(),
+  })
+);
+
+type SequenceOffset = z.infer<typeof SequenceOffset>;
+
+const useSequenceOffset = (program: string) => {
+  return useQuery({
+    queryKey: ["SequenceOffset", program],
+    queryFn: () =>
+      subWebContext(program)
+        .web.lists.getByTitle("SequenceOffset")
+        .items.top(1)
+        .orderBy("Id", false)(),
+    select: (data: SequenceOffset) => SequenceOffset.parse(data),
+    staleTime: Infinity,
+    gcTime: Infinity,
+  });
+};
+
+export const useGetSequenceNumber = (program: string) => {
+  const sequenceOffset = useSequenceOffset(program);
+
+  return useMutation({
+    mutationFn: async (title: string) => {
+      const offset = sequenceOffset.data?.[0].Title || 0;
+      const newSequenceNumber = await subWebContext(program)
```

Comment:
Was about to write a comment that I thought the new Sequence Number list is uneccessary, as you can just use the ID of the new PCOL folder minus the offset, until, I realized, that that ID will grow faster, as all files in those folders eat up IDs. Good call on this new list.